packages/runtime/src/services/LocalNode.ts

```diff
@@ -1,38 +1,67 @@
 
+import { ExecutionScopes } from '../definitions/ExecutionScope.js';
 import { createNodeFilename } from '../definitions/Files.js';
 
 import ImplementationNotFound from '../errors/ImplementationNotFound.js';
 import ProcedureNotFound from '../errors/ProcedureNotFound.js';
-import RepositoryNotAvailable from '../errors/RepositoryNotAvailable.js';
 
 import Procedure from '../models/Procedure.js';
 import Request from '../models/Request.js';
 import Response from '../models/Response.js';
 import Segment from '../models/Segment.js';
-import Module from '../types/Module.js';
 import ArgumentConstructor from '../utils/ArgumentConstructor.js';
-import ModuleLoader from '../utils/ModuleLoader.js';
 
 import Gateway from './Gateway.js';
-import LocalGateway from './LocalGateway.js';
 import Node from './Node.js';
 import Repository from './Repository.js';
 
-import { setRuntime, setDependencyLoader } from '../hooks.js';
+import { setRuntime } from '../hooks.js';
 
 export default class LocalNode extends Node
 {
+    #gateway?: Gateway;
     #argumentConstructor: ArgumentConstructor;
+
+    #segmentNames: Set<string> = new Set();
     #segments: Map<string, Segment> = new Map();
-    #gateway?: Gateway;
-    #repository?: Repository;
-    #clientId = '';
 
-    constructor(url?: string, argumentConstructor = new ArgumentConstructor())
+    constructor(repository: Repository, gateway?: Gateway, url?: string, argumentConstructor = new ArgumentConstructor())
     {
-        super(url);
+        super(repository, url);
 
+        this.#gateway = gateway;
         this.#argumentConstructor = argumentConstructor;
+
+        setRuntime(this);
+    }
+
+    set segmentNames(names: Set<string>)
+    {
+        this.#segmentNames = names;
+    }
+
+    async start(): Promise<void>
+    {
+        await super.start();
+
+        await this.#loadSegments();
+
+        if (this.#gateway !== undefined)
+        {
+            await this.#gateway.start();
+        }
+    }
+
+    async stop(): Promise<void>
```

Comment:
That's correct. I didn't implement it because it isn't required for implementing this feature, and it wasn't there before. We can make a separate issue for this, and the de-registration at the repository.